### DISCUSSION TITLE
Delayed action, camera not waking up.

### DISCUSSION THREAD

**Discussion by aske2002** [gopro/labs#95] [General]
We are doing a study of fish, using 15 hero 9's and 10 hero 7's. All with labs firmware.

For this we are using two QR-codes, one to set the framerate, time, etc, and one to start a delayed action loop. Eg.

Command: mVr4p60fW0oT220929011327 (set resolution, time and framerate)
Turn the camera on and off
Command: !1680N!S!120E!R

We deploy all the cameras, and check the cameras every day, to change batteries and sd-card, and then restart the delayed actions loop.

When the second QR-code is scanned, the camera says "Start time HH:MM:MS:MS (28 mins), press the button to cancel", and then turns off. This is expected behavior. The problem comes when the camera has to turn on again. Some of the cameras will turn on and run the loop perfectly fine, but about 50% won't even turn on the first time. This issue mostly seems to be camera specific, but also random at times. It mostly occurs for the Hero 9's, and sometimes for the Hero 7's. Sometimes the cameras will record a few videos, before just stopping. (Battery is still above 50%). We are submerging the cameras in the water, but the temperature is about 18 degrees Celsius, so that shouldn't be a problem with the batteries.

We've tried all kinds of different stuff to fix the issue, like reinstalling labs firmware, factory resets, doing things in a different order, filming a short video manually, formatting sd-cards, and more, but we can't seem to find any solution. Below is a part of the qrlog.txt from one of the faulty cameras, where it didn't record.

[2056-09-30 18:17:14] Read QR:mVr4p60fW0oT220929001934
[2022-09-29 00:19:38] Shutdown:inprogress
[2022-09-29 00:19:47] Labs:Main Start
[2022-09-29 00:19:47] Labs:Initialize
[2022-09-29 00:19:51] Read QR:!1680N!S!120E!R
[2022-09-29 00:19:57] Shutdown and Sleep:1560 seconds (28 minutes)
[2022-09-29 00:19:58] shutdown:rtc wake scheduled
[2022-09-29 01:10:13] Labs:Main Start
[2022-09-29 01:10:13] Labs:Initialize

As you can see it did not wake up, I ended up turning it on manually, it gave me a message saying it will turn on in 23.9 hours, which is the scheduled time + 24 hrs.

Any help is appreciated very much, as this issue is very hurtful for the study.

Thank you!

**Comment by dnewman-gpsw** [comment]
It does seem that some individual cameras are worse at this style of repeated action than others. For Labs testing we typically use shorter intervals, or way longer, and never (or rarely) see this issue over a wide range of models. I will run a bunch of cameras with !1680N to see is this interval is causing more trouble (3x HERO9, 2x HERO7s, 1x HERO10, 1x HERO11). So far cameras are working. This feature (hack) has never been 100% reliable, but I've personally had way more success than reports like yours (I once run a camera for a year, three photos a day at sunset.)  There should be no trick to it, although Wireless Off is common for my test cameras.
![image](https://user-images.githubusercontent.com/10856084/193091698-8a94e9b5-cf49-45b3-9225-aa7cddc53462.png)

**Reply by aske2002** [reply]
Thank you very much for testing. Wifi is turned off on all cameras.

**Reply by aske2002** [reply]
We figured it out. Some of the batteries we had bought for our cameras were apparently unoriginal, which stopped the wake function from working somehow. I am very sorry for the inconvenience.

 I hope this can help someone else if they encounter this problem.

**Reply by dnewman-gpsw** [reply]
That helps explains a lot.  All my seven testing cameras have been running your script since 9am (5.5 hours), I expect they will all run until the batteries are depleted.

**Comment by dnewman-gpsw** [comment]
Final results: 
HERO9 A - Run out of battery at 6pm (started at 10am) - 8 hours
HERO9 B - Run out of battery at 4:30pm (started at 10am) - 6.5 hours
HERO9 C - Run out of battery at 6:30pm (started at 10am) - 8.5 hours
HERO7 A - Run out of battery at 3:40pm (started at 10:10) - 5.5 hours (old battery, not looking in good condition)
HERO7 B - Run out of battery at 6:45pm (started at 12:15) - 6.5 hours
HERO10  - Run out of battery at 9pm (started at 9:30) - 11.5 hours (Enduro battery)  - Winner
HERO11  - Run out of battery at 4:30pm (started at 9:30) - 7 hours (Enduro battery - although was running it at 180Mb/s oops!)

All cameras worked.

**Reply by aske2002** [reply]
Thank you, I see the same results across our cameras now that we only use original batteries.

We're now trying to use the conditional code ">06:00<22:00!S+!120E+!1680R~!06:00R", so that we can run this loop only during daytime hours. The problem is that many of the cameras do not turn on the next day at 6:00, but are set to turn on at 6:00 the day after that. Eg. stops at 22:00 10/01/2022, no recordings from 10/02/2022, starts recording again 6:00 10/03/2022. Do you know what the problem could be here?

**Reply by dnewman-gpsw** [reply]
There is a bug in that command, ">6:00" and start at 6:00. As there is no >= command, if it wakes between 06:00 and 06:01, it will sleep until the next day. You need to do  ">05:59<22:00!S+!120E+!1680R~!06:00R"  or just simplify to ">22:00!06:00R!S!120E!1680R"
